File: core/src/com/oni/donjon/action/AbstractAction.java
package com.oni.donjon.action;

import com.badlogic.gdx.math.Rectangle;
import com.badlogic.gdx.math.Vector2;
import com.oni.donjon.data.GameData;
import com.oni.donjon.map.Tile;

/**
 * @author Daniel Chesters (on 02/06/14).
 */
public abstract class AbstractAction implements Action {
    protected boolean isPlayerSamePositionAsTile(Tile tile) {
        return testPositionBetweenTileAndPlayer(tile, 1);
    }

    protected boolean isNearPlayer(Tile tile) {
        return testPositionBetweenTileAndPlayer(tile, 1.5f);
    }

    private boolean testPositionBetweenTileAndPlayer(Tile tile, float distance) {
        Rectangle tileRectangle = tile.getRectangle();
        Vector2 playerPosition = GameData.INSTANCE.getPlayerPosition();
        return Math.abs(tileRectangle.getX() - (int) playerPosition.x) < distance
            && Math.abs(tileRectangle.getY() - (int) playerPosition.y) < distance;
    }
}

File: core/src/com/oni/donjon/action/AbstractAction.kt
package com.oni.donjon.action

import com.oni.donjon.data.GameData
import com.oni.donjon.map.Tile

/**
 * @author Daniel Chesters (on 02/06/14).
 */
abstract class AbstractAction : Action {
    protected fun isPlayerSamePositionAsTile(tile: Tile): Boolean {
        return testPositionBetweenTileAndPlayer(tile, 1f)
    }

    protected fun isNearPlayer(tile: Tile): Boolean {
        return testPositionBetweenTileAndPlayer(tile, 1.5f)
    }

    private fun testPositionBetweenTileAndPlayer(tile: Tile, distance: Float): Boolean {
        val tileRectangle = tile.rectangle
        val playerPosition = GameData.getPlayerPosition()
        return Math.abs(tileRectangle.getX() - playerPosition.x.toInt()) < distance && Math.abs(tileRectangle.getY() - playerPosition.y.toInt()) < distance
    }
}
